plutus-pab/src/Plutus/PAB/MockApp.hs

```diff
@@ -210,34 +206,54 @@ handleFinalEffects state action =
     & handleUUIDEffect
     & runM
 
+notifyChainIndex ::
+       ( Member ChainEffect effs
+       , Member MultiAgentPABEffect effs
+       )
+    => ( ChainControlEffect ~> Eff effs )
+    ->   ChainControlEffect ~> Eff effs
+notifyChainIndex handler = \case
+    ProcessBlock -> do
+        block <- handler ProcessBlock
```

Comment:
maybe [`interpose`](https://hackage.haskell.org/package/freer-simple-1.2.1.1/docs/Control-Monad-Freer.html#v:interpose) could be used instead of passing the handler

Reply:
Other than that, I think this is a good solution for dealing with the notifications in the tests 👍

Reply:
I am not sure that would work in this case, because I need the return from running the handler to get to the `block` that has been added.

Reply:
Ok, how about something like this:

```haskell
notifyChainIndex ::
       ( Member ChainEffect effs
       , Member MultiAgentPABEffect effs
       )
    => ChainControlEffect ~> Eff (ChainControlEffect ': effs)
notifyChainIndex = \case
    ProcessBlock -> do
        block <- processBlock
```

Then you could do 

```haskell
handleTopLevelEffects action =
  action
    & interpret handleControlChain
    & reinterpret notifyChainIndex
    & PAB.MultiAgent.handleMultiAgent
    & PAB.MultiAgent.handleMultiAgent
```

Reply:
This looks much better! Thanks!

Reply:
`reinterpret notifyChainIndex` should come before `interpret handleControlChain` though.